modules/auxiliary/scanner/smb/smb_enumusers_domain.rb

```diff
@@ -24,191 +25,109 @@ def initialize
-      self.simple.connect("\\\\#{simple.address}\\IPC$") # smb_login connects to this share for some reason and it doesn't work unless we do too
+      self.simple = session.simple_client
+      results = run_session
+    else
+      results = run_rhost
     end
 
-    ports.each do |port|
-
-      @rport = port
-      begin
-        unless session
-          connect()
-          smb_login()
-        end
-
-        uuid = [ '6bffd098-a112-3610-9833-46c3f87e345a', '1.0' ]
-
-        handle = dcerpc_handle_target(
-          uuid[0], uuid[1], 'ncacn_np', ["\\wkssvc"], simple.address
-        )
-        begin
-          dcerpc_bind(handle)
-          stub =
-            NDR.uwstring("\\\\" + simple.address) + # Server Name
-            NDR.long(1) +           # Level
-            NDR.long(1) +           # Ctr
-            NDR.long(rand(0xffffffff)) +  # ref id
-            NDR.long(0) +           # entries read
-            NDR.long(0) +           # null ptr to user0
-
-            NDR.long(0xffffffff) +      # Prefmaxlen
-            NDR.long(rand(0xffffffff)) +  # ref id
-            NDR.long(0)             # null ptr to resume handle
-
-          dcerpc.call(2,stub)
-
-          resp = dcerpc.last_response ? dcerpc.last_response.stub_data : nil
-
-          accounts = parse_net_wksta_enum_users_info(resp)
-          accounts.shift
-
-          if datastore['VERBOSE']
-            accounts.each do |x|
-              print_status x[:logon_domain] + "\\" + x[:account_name] +
-                "\t(logon_server: #{x[:logon_server]}, other_domains: #{x[:other_domains]})"
-            end
-          else
-            print_status "#{accounts.collect{|x| x[:logon_domain] + "\\" + x[:account_name]}.join(", ")}"
-          end
-
-          found_accounts = []
-          accounts.each do |x|
-            comp_user = x[:logon_domain] + "\\" + x[:account_name]
-            found_accounts.push(comp_user.scan(/[[:print:]]/).join) unless found_accounts.include?(comp_user.scan(/[[:print:]]/).join)
-          end
-
-          found_accounts.each do |comp_user|
-            if comp_user.to_s =~ /\$$/
-              next
-            end
-
-            print_good("Found user: #{comp_user}")
-            store_username(comp_user, resp, simple.address, rport)
-          end
-
-        rescue ::Rex::Proto::SMB::Exceptions::ErrorCode => e
-          print_error("UUID #{uuid[0]} #{uuid[1]} ERROR 0x%.8x" % e.error_code)
-          #puts e
-          #return
-        rescue ::Exception => e
-          print_error("UUID #{uuid[0]} #{uuid[1]} ERROR #{$!}")
-          #puts e
-          #return
-        end
-
-        disconnect()
-        return
-      rescue ::Exception
-        print_line($!.to_s)
-      end
+    unless results.to_s.empty?
+      results_table = format_results(results)
+      results_table.rows = results_table.rows.uniq # Remove potentially duplicate entries from port 139 & 445
+      
+      print_line
+      print_line results_table.to_s
```

Comment:
Would you be opposed to keeping this all on one line like it was before?

## Old Output
```
msf6 auxiliary(scanner/smb/smb_enumusers_domain) > run

[*] Using existing session 1
[*] MSFLAB\smcintyre, MSFLAB\smcintyre, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$, MSFLAB\DC$
[+] Found user: MSFLAB\smcintyre
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/smb/smb_enumusers_domain) >
```

## New Output
```
msf6 auxiliary(scanner/smb/smb_enumusers_domain) > run


Logged-on Users
===============

    Name       Domain  Other Domains  Logon Server
    ----       ------  -------------  ------------
    DC$        MSFLAB
    smcintyre  MSFLAB                 DC

[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/smb/smb_enumusers_domain) >
```

## Proposed Output
I think ideally, if we kept the output on one line, uniqued the names, then comma joined them in a list enclosed in square brackets and prefixed with the domain, it'd be the same output as the `auxiliary/scanner/smb/smb_enumusers` module. It's worth us considering consolidating these two modules, but for now having output that looks the same would make it pretty easy to understand.

```
[+] MSFLAB [ smcintyre, DC$ ]
```

Reply:
sure why not ^^